File: cpp_version/src/cliente.cpp
#include "cliente.hpp"

Cliente::Cliente(std::string nome, std::string cpf, Contato* contato, Endereco* endereco, int numNotaFiscal) : Pessoa(nome, cpf, contato, endereco) {
	this->modCliente(nome, cpf, contato, endereco, numNotaFiscal);
}

void Cliente::modCliente(std::string nome, std::string cpf, Contato* contato, Endereco* endereco, int numNotaFiscal) {
	this->modPessoa(nome, cpf, contato, endereco);
	this->numNotaFiscal = numNotaFiscal;
}

int Cliente::getNumNotaFiscal() {
	return this->numNotaFiscal;
}

File: cliente.py
from contato import *
from endereco import *

class Cliente:
    def __init__(self, nome, cpf, contato, endereco, numNotaFiscal):
        self.nome = nome                            #str
        self.cpf = cpf                              #str
        self.numNotaFiscal = numNotaFiscal          #int
        self.contato = contato                      #contato
        self.endereco = endereco                    #endereco

    def infoParciais(self):
        return "\n\t\tNome: " + str(self.nome) + "\n\t\tContato: " + str(self.contato.infoCompleta()) + "\n\t\tEndereco: " + str(self.endereco) + "\n\t\tNota Fiscal: " + str(self.numNotaFiscal)

    def infoCompleta(self):
        return "Info: \n\t\tNome: " + str(self.nome) + "\n\tContato: " + str(self.contato.infoCompleta()) + "\n\t\tEndereco: " + str(self.endereco.infoCompleta()) + "\n\tNota Fiscal: " + str(self.numNotaFiscal) + "\n\t\tCpf: " + str(self.cpf)
